src/openrct2/interface/InteractiveConsole.cpp

```diff
@@ -878,17 +880,17 @@ static int32_t cc_set(InteractiveConsole& console, const arguments_t& argv)
             }
             else
             {
-                for (i = 0; i < CLIMATE_COUNT; i++)
+                for (i = 0; i < static_cast<uint8_t>(ClimateType::Count); i++)
                 {
                     if (argv[1] == ClimateNames[i])
                     {
-                        gClimate = i;
+                        gClimate = ClimateType{ static_cast<uint8_t>(i) };
                         break;
                     }
                 }
             }
 
-            if (i == CLIMATE_COUNT)
+            if (i == static_cast<uint8_t>(ClimateType::Count))
```

Comment:
This code is odd. This whole function is wrong. Sorry could you fix it. It should be the following:
uint8_t newClimate = static_cast<uint8_t>(ClimateType::Count);
If `int_valid[0]` then `newClimate =  int_val[0]` else loop through comparing argv[1] and ClimateNames to find a newClimate.

After that if else branch validate the `newClimate` if it is >= ClimateType::Count write error message from L871 and sent invalidArgs to true. Else issue a `ClimateSetAction` with the new climate and finally do the console.Eecute from L894.